crates/polars-core/src/chunked_array/ops/rolling_window.rs

```diff
@@ -168,7 +168,8 @@ mod inner_mod {
                         }
                         // ensure the length is correct
                         series_container._get_inner_mut().compute_len();
-
+                        // reset flags as we reuse this container
+                        series_container.clear_settings();
```

Comment:
We reuse the following series to avoid re-allocation.

https://github.com/pola-rs/polars/blob/9bfa5b6db9b112328c657651c2def6543ae46473/crates/polars-core/src/chunked_array/ops/rolling_window.rs#L93-L94

But it is `Ascending` because when we first computed its size, we went to the following branch:

https://github.com/pola-rs/polars/blob/9bfa5b6db9b112328c657651c2def6543ae46473/crates/polars-core/src/chunked_array/ops/chunkops.rs#L77-L79

This will affect subsequent function call. Maybe simply unset the sorting mark when creating it can solve this problem. But for safety reasons, I chose to reset all states before each function call. WDYT?

Reply:
This is indeed the way to go. Only the `clear_settings` should be before the `compute_len()`. As `compute_len` sets sorted info.

Reply:
Ah, that's a foolish mistake, I messed up their order. :)

Reply:
Order exchanged.